spec/index.html

```diff
@@ -2881,8 +2898,15 @@ <h3>Insecure Hash Algorithms</h3>
       and implementations of it can be parameterized to use a different
       hash function, without the need to make any changes to the
       canonicalization algorithm itself.
-      However, using a different hash algorithm will generally lead to different results.
+      However, using a different hash algorithm will generally lead to different results;
+      applications making use of this specification should carefully weigh the advantages
+      and disadvantages of using an alternative hash function.
     </p>
+
+    <p class="note">The possible implications of the default hash algorithm
+      becoming insecure are mitigated by that fact that no internal hash
+      values are revealed, and the algorithms are designed to cope
```

Comment:
Something like this reads a better to me.  No confusion on hash vs canonicalization algorithm.  And I think this can be singular?
```suggestion
      values are revealed, and the canonicalization algorithm is designed to cope
```